javascript/diff_match_patch_uncompressed.js

```diff
@@ -601,6 +619,9 @@ diff_match_patch.prototype.diff_commonSuffix = function(text1, text2) {
     }
     pointermid = Math.floor((pointermax - pointermin) / 2 + pointermin);
   }
+  if (insideSurrogate(text1, text1.length - pointermid)) {
+    pointermid--;
+  }
```

Comment:
I came up with this instead, but I don't have a test that fails in either case so obviously I'm missing a test case

```js
if (pointer mid < length - 1 && this.isLowSurrogate(text1[poitnermid])) {
  pointermid++;
}
```

my reasoning was that if the common suffix _starts_ on a low surrogate then we have to bump it to the right so that the surrogate half doesn't end up in the common part - backwards from `commonPrefix` where we want to shift to the left in order to avoid the same for high surrogates